pkg/sql/compile/ddl.go

```diff
@@ -90,6 +89,29 @@ func (s *Scope) CreateTable(c *Compile) error {
 	return colexec.CreateAutoIncrCol(dbSource, c.ctx, c.proc, planCols, tblName)
 }
 
+func (s *Scope) TruncateTable(c *Compile) (uint64, error) {
+	tqry := s.Plan.GetDdl().GetTruncateTable()
+	dbName := tqry.GetDatabase()
+	dbSource, err := c.e.Database(c.ctx, dbName, c.proc.TxnOperator)
+	if err != nil {
+		return 0, err
+	}
+	tblName := tqry.GetTable()
+	var rel engine.Relation
+	if rel, err = dbSource.Relation(c.ctx, tblName); err != nil {
+		return 0, err
+	}
+	affect, err := rel.Truncate(c.ctx)
+	if err != nil {
+		return 0, err
+	}
+	err = colexec.ResetAutoInsrCol(rel, dbSource, c.ctx, c.proc, rel.GetTableID(c.ctx))
```

Comment:
The rel.GetTableId get new id or retrieve the id that was already generated somewhere?